### ISSUE TITLE
Customizable messages on break window

### ISSUE THREAD

**Issue by AllanChain** [AllanChain/sane-break#12]
See also #3 

Customizable messages can be useful. While it's impossible to have the default messages suitable for everyone, it can be set to remind us of exactly things we personally need reminders of.

Maybe we can borrow some from Stretchly ([License compatible](https://www.gnu.org/licenses/license-list.en.html#FreeBSD)):
- https://github.com/hovancik/stretchly/blob/692415cbb654281c2c997d4a99cd4f58f23e4339/app/utils/defaultBreakIdeas.js#L7
- https://github.com/hovancik/stretchly/blob/692415cbb654281c2c997d4a99cd4f58f23e4339/app/utils/defaultMicrobreakIdeas.js

Considering this is a nice-to-have feature and being relatively easy to implement, we put it as medium priority.

**Comment by niiinjoe** [comment]
+1

Coming from SafeEyes, I miss its break suggestion, especially on small break that usually suggest eyes relaxation.

**Comment by wolftune** [comment]
btw, I have a bunch of notes about what messages could be (drawing on other programs and on mindful-reminder lists), so I'd be happy to participate in deciding on a default starting set when this feature is getting added

**Comment by AllanChain** [comment]
I'm wondering if it's a good idea to provide default messages. The major concern is that we have to maintain the translation for all the messages, and this raises problems for accepting user contributions. I don't know whether there will be enough users benefiting from the default list.

Anyway, the first step is to add a text box in the preference UI, allowing users to set their own messages for small and big breaks.

**Comment by AllanChain** [comment]
> So, for this issue, the question is mostly the format. 

Why not a simple plain text, one line per message? Qt [supports basic Markdown formatting](https://doc.qt.io/qt-6/qt.html#TextFormat-enum), and it's possible to have some formats. Users just need to copy and paste them into a multi line textarea in the preference UI. I think this is the most straightforward way.

**Comment by wolftune** [comment]
Yes, plain text with one line per message is good. Perhaps ignore empty lines so people can space out the messages if they prefer to view the list that way? I like the idea of supporting basic markdown formatting so that basic styles can be used like to highlight certain words by making them bold or italic etc.

**Comment by niiinjoe** [comment]
I vote for a simple text area with one line per message (for each type of break). Sequential or random order is both fine for me.

For the defaults, how about a single message for each break instead of "Time for a small break" and "Time for a big break"? Something like "Please take a little eyes break" or "Please take some rest or walk for a while" or something else that is not just "Time for ..." 😄  Just two more messages shouldn't increase the translation load much.

**Comment by wolftune** [comment]
I see that https://github.com/AllanChain/sane-break/commit/e12ce63982120a8afc68d0a22aab27e60219fdef added custom messages just for the prompt. I guess that makes some sense, though it seems a little odd to me (though I can't know all the ways people around the world might want this to work). Customizing this seems sensible, not sure if it's overkill to offer a list of variable messages just for the prompt though.

I think messages for the full-screen break are the more valuable idea, and distinct for small and big.

**Comment by AllanChain** [comment]
> I think messages for the full-screen break are the more valuable idea, and distinct for small and big.

I think a customizable message and adding a separate customizable message on full-screen serves the same purpose, with the latter being more verbose. The difference between a small and big break is already in the flash color, and I don't think we have to keep the "timer for a small/big break" message. One customizable message for both prompt and full-screen window is better, I think.

**Comment by wolftune** [comment]
I'm not sure I understand the reply. If you notice the sorts of messages Stretchly and others use, some of them are far too long to put in the prompt and are in no way prompt-appropriate.

I guess what you are saying is to have *separate* prompt messages and full-screen messages… but to have identical text for small and big breaks?

But people who like these instructional messages really do use different things for small and big breaks. Like small might say "close your eyes" and big might be "take a walk outside", and those are not interchangeable.

And I don't think it's good to rely on color alone to identify whether the prompt is for small or big. It's nice to be labeled small or big for the *prompt* at least.
